pages/hooks/usereducer.mdx

```diff
@@ -2,6 +2,78 @@
 title: useReducer
 ---
 
+import useReducer from '../../examples/files/hooks/useReducer.js'
+import useReducerRefactor from '../../examples/files/hooks/useReducerRefactor.js'
+import useReducerLazyInit from '../../examples/files/hooks/useReducerLazyInit.js'
+
 # useReducer
 
-Coming soon!
+useReducer is similar to useState since it allows us to control state inside a function component.  
+The difference between the two is that useReducer allows us to handle multiple states and control complex state changes. 
+
+The useReducer hook takes two mandatory arguments: a reducer function and an initial argument. It can also take in an optional third argument called the initializer, which we will discuss later. useReducer returns the current state and a dispatch function.  
+
+`const [state, dispatch] = useReducer(reducer, initialState)`
+
+## Reducer
+
+The reducer function is a pure function that takes in the previous state and an action. It returns a new state. 
+
+```
+const reducer = (prevState, action) => {
+  // do action
+  return newState
+}
+const [state, dispatch] = useReducer(reducer, initialState)
+```
+
+## Dispatch
+
+The dispatch function, when called, passes the action to the reducer.
+
+In the Example code below, the initial state is `No Pet Selected`. When we click the `cat` radio button, dispatch is called with `{type: 'cat'}` as its argument. This argument gets passed to the reducer as `action`. Then the reducer follows the logic `if (action.type === 'cat')` and returns the new state `You selected cat`
+
+<Example code={useReducer}/>
+
+## useState and useReducer
+
+Here we will highlight some of the differences between useState and useReducer. 
+
+Recall our example in useEffect, there were multiple declarations of useState and several areas where state was changed.  
+We can refactor the code with `useReducer` and handle all state changes in one `reducer` function. 
+
+<Example code={useReducerRefactor}/>
+
+There are situations when one hook is the preferable choice over the other. 
+
+For simpler state transition that use JavaScript primitives, it would be ideal to use useState. 
+
+When there are different properties that are dependent on one another, JavaScript objects or arrays as state, or complex state business logic, it is recommended to useReducer because state will be managed in one place and it would provide a more predictable state architecture. 
+
+## The third argument: Initializer
+
+useReducer allows us to initialize state in two different ways.  
```

Comment:
I think if we want to show an example for the initializer arg, we need to think of an example with a computationally expensive function, since if people were to copy this example directly as a pattern, it wouldn't be the best.

If we can't think of a good one, we could probably get away with just mentioning the 3rd arg and linking to the react docs.

Reply:
Sounds good to me! 👍